.github/workflows/check-links/check-links.sh

```diff
@@ -0,0 +1,40 @@
+#!/bin/bash
+
+# Copyright The OpenTelemetry Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -eu
+
+SCRIPT_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+REPO_DIR="$( cd "${SCRIPT_DIR}/../../../" && pwd )"
+GITHUB_REF=${GITHUB_REF:-}
+
+diff_files="$( git diff HEAD origin/master --name-only )"
+check_all_files=1
+if [[ "$GITHUB_REF" = "ref/heads/master" ]] || [[ -n "$( echo "$diff_files" | grep ".github/workflows/check-links" )" ]]; then
```

Comment:
copying over the contents of `.circleci/check-links/`. Needed to tweak this script so it works with the GitHub Actions environment variables and container structure